src/react-components/presence-log.js

```diff
@@ -96,16 +106,27 @@ class PresenceLog extends Component {
         );
       }
     }
-  };
-
-  render() {
-    const presenceClasses = {
-      [styles.presenceLog]: true,
-      [styles.presenceLogInRoom]: this.props.inRoom
-    };
+  }, [history, hubId, preset, intl, onViewProfile, presences, include, exclude, isMod]);
 
-    return <div className={classNames(presenceClasses)}>{this.props.entries.map(this.domForEntry)}</div>;
-  }
+  useEffect(() => {
+    setLogEntries(entries.map(domForEntry));
```

Comment:
whats the goal with this effect and logEntries state vs just using entries directly?

Reply:
Not sure if I get this one. Entries is the raw message data, logEntries are the DOM elements processed via map.

Reply:
Right. But why have an effect and state instead of mapping in render like we were doing before. In either case we still have to render whenever a new entry comes in, so this just seems like an added layer of indirection.